main.py

```diff
@@ -215,23 +214,12 @@ def _switch_colors(self) -> dict:
                                 )}
             return colors
 
-    # pylint: disable=missing-timeout
-    def _send_flow_mods(self, dpid_flows: dict) -> None:
-        """Send FlowMods."""
-        for dpid, flows in dpid_flows.items():
-            res = requests.post(
-                self._flow_manager_url % dpid,
-                json={'flows': flows, 'force': True}
-            )
-            if res.status_code // 100 != 2:
-                log.error(f'Flow manager returned an error inserting '
-                          f'flows {flows}. Status code {res.status_code} '
-                          f'on dpid {dpid}')
-
-    def _remove_flow_mods(self, flows: dict, force: bool = True) -> None:
+    def _send_flow_mods(
+        self, flows: dict, action: str, force: bool = True
+    ) -> None:
         """Remove FlowMods"""
         for dpid, mod_flows in flows.items():
-            name = "kytos.flow_manager.flows.delete"
+            name = f"kytos.flow_manager.flows.{action}"
```

Comment:
@Alopalao, can you also change the to use the "single" flows handler?

`kytos.flow_manager.flows.single.(install|delete)`
 
That handler executes in a single thread, so there's never a concern if a installation/deletion of out of order resume/preemption type of bugs. as it is, here it would still work, but since flows can be deleted when link gets disabled, let's not leave room for that to potentially happen and since you're already at it. And, coloring doesn't send too many flows in different events in normal circumstances so using the single thread handler the IO throughput should still be decent. See what I mean? So, please, also quickly explore trying to disable/enable the link and confirm it's working well. After that it can be shipped.

Reply:
Changed